app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -439,43 +444,58 @@ MinionPoller = {
     var isWorker = MinionPoller.selectedNodes.indexOf(minion.id) !== -1;
     var isUnused = !isMaster && !isWorker;
 
-    var masterHtml;
     var masterChecked = '';
-    var minionHtml;
     var minionChecked = '';
+    var disabledHtml = '';
+    if (State.pendingRemovalMinionId != null){
+      disabledHtml = 'disabled="disabled"';
+    }
 
     if (isMaster) {
       masterChecked = 'checked';
     }
 
-    masterHtml = '\
-      <input name="roles[master][]" id="roles_master_' + minion.id +
+    var masterHtml = '\
+      <input hidden name="roles[master][]" id="roles_master_' + minion.id +
```

Comment:
Any reason to add this `hidden` attribute?

Reply:
When disabling the buttons a checkbox became visible.